pkg/core/input.go

```diff
@@ -6,14 +6,11 @@ import (
 )
 
 type Input interface {
-	Start(emitter Emitter, router Router) error
+	NewPositionCache() (position_store.PositionCacheInterface, error)
+	Start(emitter Emitter, router Router, positionCache position_store.PositionCacheInterface) error
```

Comment:
this seems strange. If `Input` is the only component able to create position cache, then `Start` should not inject it.
The other way is, `PositionCache` is created by `server`, and injected into `Input`, then there shouldn't have `NewPositionCache` method.